vms/platformvm/state/diff.go

```diff
@@ -504,6 +572,26 @@ func (d *diff) Apply(baseState Chain) error {
 			baseState.DeleteExpiry(entry)
 		}
 	}
+	// Ensure that all sov deletions happen before any sov additions. This
+	// ensures that a subnetID+nodeID pair that was deleted and then re-added in
+	// a single diff can't get reordered into the addition happening first;
+	// which would return an error.
+	for _, sov := range d.sovDiff.modified {
```

Comment:
Playing devil's advocate here, we're iterating here over a map, and calling `baseState.PutSubnetOnlyValidator` which may return with different errors.  This means that across different nodes, we may fail differently, no? 

Is that a concern?

Reply:
It is generally unexpected for `Apply` to ever error.

`Apply` is called during `Accept`. If `Apply` were to return an error, that error would be propagated through `Accept` into the consensus engine (causing a FATAL).